src/java.desktop/share/classes/java/awt/Graphics2D.java

```diff
@@ -1200,10 +1200,14 @@ public abstract boolean hit(Rectangle rect,
      * previously been set, or if the clip has been cleared using
      * {@link Graphics#setClip(Shape) setClip} with a {@code null}
      * argument, the specified {@code Shape} becomes the new
-     * user clip.
+     * user clip. Since this method intersects the specified shape
+     * with the current clip, it will throw {@code NullPointerException} for a null shape
+     * unless the user clip is also null.
```

Comment:
Shall the text be re-wrapped?

I'm not sure if null should always be {@code null}, it's not used consistently. At the same time, two lines above it's “…with a {@code null} argument…”; it makes sense to make Javadoc consistent in this regard at least for one method.

Reply:
ok .fair enough...done..